Here's the commented version of the code snippet:

```c++
// Copyright 2019 gRPC authors.

// Header guard to prevent multiple inclusions of this header file
#ifndef GRPCPP_IMPL_CODEGEN_CLIENT_CALLBACK_H
#define GRPCPP_IMPL_CODEGEN_CLIENT_CALLBACK_H

// Includes the main client callback support functionality from gRPC
// This provides the core implementations for client-side callback mechanisms
#include <grpcpp/support/client_callback.h>

// End of header guard
#endif
```

The comments explain:
1. The purpose of the header guard (a standard C++ practice to prevent multiple inclusions)
2. What the included file provides (core client callback functionality)
3. The closing of the header guard

The file appears to be a thin wrapper that primarily serves to include the actual implementation from the support directory while maintaining proper header guard protection.